### ISSUE TITLE
useless_attribute should not warn unreachable_pub

### ISSUE THREAD

**Issue by taiki-e** [rust-lang/rust-clippy#4106]
`unreachable_pub` can apply to `use`.

```rust
#![warn(unreachable_pub)]
#![warn(clippy::useless_attribute)]

mod a {
    mod b {
        #[allow(unreachable_pub)]
        pub struct C {}
    }
    #[allow(unreachable_pub)]
    pub use b::C;
}
```

[Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=12bdb8024c99ae984f76ef00372b4404)
